diff --git a/i3lock.c b/i3lock.c
index cf1526d..c9d7f7f 100644
--- a/i3lock.c
+++ b/i3lock.c
@@ -655,8 +655,11 @@ static void xcb_check_cb(EV_P_ ev_check *w, int revents) {
                     dont_fork = true;
 
                     /* In the parent process, we exit */
-                    if (fork() != 0)
+                    int pid = fork();
+                    if (pid != 0) {
+                        printf("%d", pid);
                         exit(0);
+                    }
 
                     ev_loop_fork(EV_DEFAULT);
                 }
