From c48d3f042cf392233c9b61dfe32f40b75a51395a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Benno=20F=C3=BCnfst=C3=BCck?= <benno.fuenfstueck@gmail.com>
Date: Sat, 14 Nov 2015 21:18:41 +0100
Subject: [PATCH] fix mpd crash on invalid utf8 stream title

---
 src/tag/TagString.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tag/TagString.cxx b/src/tag/TagString.cxx
index d30a07a..bdb32b9 100644
--- a/src/tag/TagString.cxx
+++ b/src/tag/TagString.cxx
@@ -41,9 +41,9 @@ FindInvalidUTF8(const char *p, const char *const end)
 		/* now call the other SequenceLengthUTF8() overload
 		   which also validates the continuations */
 		const size_t t = SequenceLengthUTF8(p);
-		assert(s == t);
 		if (t == 0)
 			return p;
+		assert(s == t);
 
 		p += s;
 	}
-- 
2.5.3

