diff --git a/eggtraymanager.c b/eggtraymanager.c
index 576e10d..fd80bff 100644
--- a/eggtraymanager.c
+++ b/eggtraymanager.c
@@ -199,10 +199,27 @@ egg_tray_manager_plug_removed (GtkSocket       *socket,
   return FALSE;
 }
 
+static gboolean
+check_socket_window_not_null(void* ignored_window, void* socket, void* manager) {
+  if(!gtk_socket_get_plug_window(GTK_SOCKET(socket))) {
+    g_object_set_data (G_OBJECT (socket), "egg-tray-child-window", NULL);
+    g_signal_emit (EGG_TRAY_MANAGER(manager), manager_signals[TRAY_ICON_REMOVED], 0, GTK_SOCKET(socket));
+    gtk_widget_destroy(GTK_SOCKET(socket));
+    return TRUE;
+  }
+  return FALSE;
+}
+
 static void
 egg_tray_manager_handle_dock_request (EggTrayManager       *manager,
 				      XClientMessageEvent  *xevent)
 {
+  /* Check if we already added this socket */
+  if(g_hash_table_lookup (manager->socket_table, GINT_TO_POINTER (xevent->data.l[2]))) return;
+
+  /* Remove sockets that are now invalid */
+  g_hash_table_foreach_remove(manager->socket_table, check_socket_window_not_null, manager);
+
   GtkWidget *socket;
   Window *window;
   
@@ -231,8 +248,10 @@ egg_tray_manager_handle_dock_request (EggTrayManager       *manager,
 
       g_hash_table_insert (manager->socket_table, GINT_TO_POINTER (xevent->data.l[2]), socket);
     }
-  else
+  else {
+    g_signal_emit (manager, manager_signals[TRAY_ICON_REMOVED], 0, socket);
     gtk_widget_destroy (socket);
+  }
 }
 
 static void
diff --git a/main.c b/main.c
index be4541b..f4c897d 100644
--- a/main.c
+++ b/main.c
@@ -30,6 +30,7 @@ tray_added (EggTrayManager *manager, GtkWidget *icon, void *data)
 {
     GtkWidget *box = (GtkWidget *)data;
 
+    gtk_widget_set_size_request(icon, 20, 20);
     gtk_box_pack_end (GTK_BOX (box), icon, FALSE, FALSE, 0);
     gtk_widget_show (icon);
 }
@@ -37,7 +38,7 @@ tray_added (EggTrayManager *manager, GtkWidget *icon, void *data)
 static void
 tray_removed (EggTrayManager *manager, GtkWidget *icon, void *data)
 {
-
+  gtk_container_remove(GTK_BOX (data), icon);
 }
 
 static void
